llama_index/node_parser/relational/markdown_element.py

```diff
@@ -59,8 +61,15 @@ def get_nodes_from_node(self, node: TextNode) -> List[BaseNode]:
         return self.get_nodes_from_elements(elements)
 
     def extract_elements(
-        self, text: str, table_filters: Optional[List[Callable]] = None, **kwargs: Any
+        self,
+        text: str,
+        table_filters: Optional[List[Callable]] = None,
+        **kwargs: Any,
     ) -> List[Element]:
+        # get node id for each node so that we can avoid using the same id for different nodes
+        node_id = None
```

Comment:
Should node_id just be in the function def?

Reply:
you mean pass as a parameter for the function?
I think it will cause typing error. since the base class has ```@abstractmethod
    def extract_elements(self, text: str, **kwargs: Any) -> List[Element]:
        """Extract elements from text."""```

Reply:
we already have table_filters, which is not in the base method

The **kwargs in the base method lets you add more 👍🏻